src/functional/iterator/embedded.py

```diff
@@ -882,34 +924,38 @@ def get_ordered_indices(
                 res.append(elem[sparse_position_tracker[axis.value]])
                 sparse_position_tracker[axis.value] += 1
             else:
-                assert isinstance(elem, (int, np.integer, slice))
+                assert isinstance(elem, (int, np.integer, slice, range))
                 res.append(elem)
     return tuple(res)
 
 
-def _tupsum(a, b):
-    def combine_slice(s, t):
-        is_slice = False
-        if isinstance(s, slice):
-            is_slice = True
-            first = 0 if s.start is None else s.start
-            assert s.step is None
-            assert s.stop is None
-        else:
-            assert isinstance(s, numbers.Integral)
-            first = s
-        if isinstance(t, slice):
-            is_slice = True
-            second = 0 if t.start is None else t.start
-            assert t.step is None
-            assert t.stop is None
-        else:
-            assert isinstance(t, numbers.Integral)
-            second = t
-        start = first + second
-        return slice(start, None) if is_slice else start
+def _shift_range(
```

Comment:
Again, here we could definie typing `overload`s to get finer grain type checking, since the output types depend only on the input types in a simple way.

Reply:
done